src/transformers/models/textnet/convert_textnet_original_to_pytorch.py

```diff
@@ -0,0 +1,202 @@
+
+    neck_in_channels = []
+    neck_out_channels = []
+    neck_kernel_size = []
+    neck_stride = []
+    neck_dilation = []
+    neck_groups = []
+
+    for i in range(1, 5):
+        layer_key = f"reduce_layer{i}"
+        layer_dict = config_dict["neck"].get(layer_key)
+
+        if layer_dict:
+            # Append values to the corresponding lists
+            neck_in_channels.append(layer_dict["in_channels"])
+            neck_out_channels.append(layer_dict["out_channels"])
+            neck_kernel_size.append(layer_dict["kernel_size"])
+            neck_stride.append(layer_dict["stride"])
+            neck_dilation.append(layer_dict["dilation"])
+            neck_groups.append(layer_dict["groups"])
+
+    textnet_config = TextNetConfig(
+        stem_kernel_size=config_dict["first_conv"]["kernel_size"],
+        stem_stride=config_dict["first_conv"]["stride"],
+        stem_num_channels=config_dict["first_conv"]["in_channels"],
+        stem_out_channels=config_dict["first_conv"]["out_channels"],
+        stem_act_func=config_dict["first_conv"]["act_func"],
+        conv_layer_kernel_sizes=[
+            backbone_config["stage1"]["kernel_size"],
+            backbone_config["stage2"]["kernel_size"],
+            backbone_config["stage3"]["kernel_size"],
+            backbone_config["stage4"]["kernel_size"],
+        ],
+        conv_layer_strides=[
+            backbone_config["stage1"]["stride"],
+            backbone_config["stage2"]["stride"],
+            backbone_config["stage3"]["stride"],
+            backbone_config["stage4"]["stride"],
+        ],
+        hidden_sizes=[
+            config_dict["first_conv"]["out_channels"],
+            backbone_config["stage1"]["out_channels"][-1],
+            backbone_config["stage2"]["out_channels"][-1],
+            backbone_config["stage3"]["out_channels"][-1],
+            backbone_config["stage4"]["out_channels"][-1],
+        ],
+        out_features=["stage1", "stage2", "stage3", "stage4"],
+        out_indices=[1, 2, 3, 4],
+        image_size=size,
+    )
+
+    return textnet_config
+
+
+def convert_textnet_checkpoint(checkpoint_url, checkpoint_config_url, pytorch_dump_folder_path):
+    response = requests.get(checkpoint_config_url)
+    content = response.text
+    namespace = {}
+
+    exec(content, namespace)
+
+    model_config = namespace.get("model")
+    data_config = namespace.get("data")
+    size = 640
+    if "train" in data_config:
+        if "short_size" in data_config["train"]:
+            size = data_config["train"]["short_size"]
+
+    if "tiny" in model_config["backbone"]["config"]:
+        config = prepare_config(tiny_config_url, size)
+    elif "small" in model_config["backbone"]["config"]:
+        config = prepare_config(small_config_url, size)
+    else:
+        config = prepare_config(base_config_url, size)
+
+    model = TextNetBackbone(config)
+    textnet_image_processor = CLIPImageProcessor(
```

Comment:
could you point to the original code where CLIP's image processing settings are being used?

Reply:
I will add an assertion in this script, to validate the image processor created with the original implementation.

Reply:
Done .